src/sentry/spans/consumers/process/factory.py

```diff
@@ -158,12 +163,58 @@ def _format_event_id(payload: Mapping[str, Any]) -> str:
     return ""
 
 
+def _process_relay_span_v1(relay_span: Mapping[str, Any]) -> MutableMapping[str, Any]:
+    snuba_span: MutableMapping[str, Any] = dict(relay_span)
+    snuba_span["exclusive_time_ms"] = int(relay_span.get("exclusive_time", 0))
+    snuba_span["parent_span_id"] = relay_span.get("parent_span_id", "0")
+    snuba_span["segment_id"] = relay_span.get("segment_id", "0")
+    snuba_span["span_id"] = relay_span.get("span_id", "0")
+    snuba_span["tags"] = {
+        k: str(v) for k, v in (relay_span.get("tags", {}) or {}).items() if v is not None
+    }
+
+    if (description := relay_span.get("description")) is not None:
+        snuba_span["description"] = description
+
+    span_data: Mapping[str, Any] = relay_span.get("user_tags", {})
+    sentry_tags: MutableMapping[str, Any] = {}
+
+    if span_data:
+        for relay_tag, snuba_tag in TAG_MAPPING.items():
+            tag_value = span_data.get(relay_tag)
+            if snuba_tag == "group":
+                if tag_value is None:
+                    metrics.incr("spans.missing_group")
+                else:
+                    try:
+                        # Test if the value is valid hexadecimal.
+                        _ = int(tag_value, 16)
+                        # If valid, set the raw value to the tag.
+                        sentry_tags["group"] = tag_value
+                    except ValueError:
+                        metrics.incr("spans.invalid_group")
+            elif tag_value is not None:
+                sentry_tags[snuba_tag] = tag_value
+
+    snuba_span["data"] = sentry_tags
+
+    _process_group_raw(
+        snuba_span=snuba_span,
+        transaction=span_data.get("transaction", ""),
+    )
+
+    return snuba_span
+
+
 def _process_message(message: Message[KafkaPayload]) -> KafkaPayload:
     payload = msgpack.unpackb(message.payload.value)
-    relay_span = payload["span"]
-    relay_span["project_id"] = payload["project_id"]
-    relay_span["event_id"] = _format_event_id(payload)
-    snuba_span = _build_snuba_span(relay_span)
+    if payload.get("version") == SPAN_SCHEMA_V1:
+        snuba_span = _process_relay_span_v1(payload)
+    else:
+        relay_span = payload["span"]
+        relay_span["project_id"] = payload["project_id"]
+        relay_span["event_id"] = _format_event_id(payload)
+        snuba_span = _process_relay_span_v0(relay_span)
```

Comment:
Is my understanding correct that the 2 versions are needed to perform the switchover in a seamless manner. Once the switchover is done, the older version can be gotten rid of?

Reply:
Yes, that's correct.